java/org/contikios/cooja/Simulation.java

```diff
@@ -53,13 +51,27 @@
  *
  * @author Fredrik Osterlind
  */
-public class Simulation extends Observable implements Runnable {
+public class Simulation extends Observable {
+
+  /* Internal marker for notifying simulation thread to start running simulation */
+  private static final Runnable START = () -> {};
+  /* Internal marker for notifying simulation thread to stop running simulation */
+  private static final Runnable STOP = () -> {};
+  /* Internal marker for notifying simulation thread to quit */
+  private static final Runnable QUIT = () -> {};
+
   public static final long MICROSECOND = 1L;
   public static final long MILLISECOND = 1000*MICROSECOND;
 
+  private final Object lock = new Object();
```

Comment:
Add a javadoc about what the lock is meant to protect.

Reply:
Good point. I will add a comment.